### PR TITLE
DOM document should store Invalid end tag `</`

### LINKED ISSUE #886: DOM document should store Invalid end tag `</`
While working on https://github.com/eclipse/lemminx/pull/881 I notice that it was very hard to support usecase with invalid end tag `</` (it's hard for instance to detect it to eat it for instance with code action).

The main problem is that DOM document doesn't stores this invalid end tag. 

We can see this problem for instance if you have this XML:

```xml
<a>
  </
</a>
```

The outline doesn't show the `</` (although it shows start tag like `<` and if you format this XML you loose the `</` content.

### PR DESCRIPTION
**Pull request by angelozerr**
DOM document should store Invalid end tag `</`

Fixes #886

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
This PR will be very helpfull to fix some error range highlight and some code actions. For the moment the invalid end tag support is used for symbol (outline) and formatting.

Here a demo with outline and formatting:

InvalidEndTag

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLCompletions.java [hunk 1]
```diff
@@ -498,7 +498,7 @@ public AutoCloseTagResponse doTagComplete(DOMDocument xmlDocument, Position posi
 			}
 		} else if (cBefore == '<' && c == '/') { // Case: <a> </|
 			DOMNode node = xmlDocument.findNodeBefore(offset);
-			while (node != null && node.isClosed()) {
+			while ((node != null && node.isClosed()) || (node.isElement() && ((DOMElement) node).isOrphanEndTag())) {
```
Comment (datho7561) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLCompletions.java:501]: I think you are planning on addressing this in the other PR, but:

```xml
<a></</a>
```
The code action puts:
```xml
<a></a</a>
```

**Review by datho7561 — CHANGES_REQUESTED** [review]

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLCompletions.java (hunk 1 above)
Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLCompletions.java:501]: Please create an issue about this usecase and please suggest what you would to generate.

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMElement.java [hunk 2]
```diff
@@ -77,6 +77,17 @@ public String getTagName() {
 		return tag;
 	}
 
+	/**
+	 * Returns true if the DOM element have a tag name and null otherwise (ex : '<'
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMElement.java]: fixed

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLCompletions.java (hunk 1 above)
Comment (datho7561) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLCompletions.java:501]: Okay. It make sense to me to move this to a separate issue. I opened https://github.com/eclipse/lemminx/issues/889

**Review by datho7561 — APPROVED** [review]

**Pull request merged**